nsqd/diskqueue.go

```diff
@@ -261,6 +266,14 @@ func (d *diskQueue) readOne() ([]byte, error) {
 		return nil, err
 	}
 
+	if msgSize < d.minMsgSize || msgSize > d.maxMsgSize {
```

Comment:
Alright, you've sold me - all of this falls under "reasonable behavior" without _actually_ knowing if it's corrupt.